pkg/internal/checkup/vmi.go

```diff
@@ -53,15 +53,23 @@ const (
 	terminationGracePeriodSeconds = 0
 )
 
-func newVMIUnderTest(name string, checkupConfig config.Config) *kvcorev1.VirtualMachineInstance {
+func newVMIUnderTest(name string, checkupConfig config.Config, configMapName string) *kvcorev1.VirtualMachineInstance {
+	const (
+		configDiskSerial = "DEADBEEF"
+		configVolumeName = "vmi-under-test-config"
+	)
+
 	optionsToApply := baseOptions(checkupConfig)
 
 	optionsToApply = append(optionsToApply,
 		vmi.WithAffinity(Affinity(checkupConfig.VMUnderTestTargetNodeName, checkupConfig.PodUID)),
 		vmi.WithSRIOVInterface(eastNetworkName, checkupConfig.VMUnderTestEastMacAddress.String(), config.VMIEastNICPCIAddress),
 		vmi.WithSRIOVInterface(westNetworkName, checkupConfig.VMUnderTestWestMacAddress.String(), config.VMIWestNICPCIAddress),
 		vmi.WithContainerDisk(rootDiskName, checkupConfig.VMUnderTestContainerDiskImage),
-		vmi.WithCloudInitNoCloudVolume(cloudInitDiskName, CloudInit(config.VMIUsername, config.VMIPassword, nil)),
+		vmi.WithCloudInitNoCloudVolume(cloudInitDiskName,
+			CloudInit(config.VMIUsername, config.VMIPassword, vmiUnderTestBootCommands(configDiskSerial))),
+		vmi.WithConfigMapVolume(configVolumeName, configMapName),
+		vmi.WithConfigMapDisk(configVolumeName, configDiskSerial),
```

Comment:
This part could potentially be moved to `baseOptions()` so it could be shared by the two VMIs.

Reply:
I thought about it, but ended up not liking the fact that part of the configmap is loaded on the base function, and part here. It makes it less understood. WDYT?